src/test/java/com/fasterxml/jackson/core/BaseTest.java

```diff
@@ -451,11 +451,15 @@ protected String readAndWrite(JsonFactory f, JsonParser p) throws IOException
         g.disable(JsonGenerator.Feature.AUTO_CLOSE_JSON_CONTENT);
         try {
             while (p.nextToken() != null) {
+                System.err.println(p.currentToken() + "  " + p.currentName() + "  " + p.currentValue());
```

Comment:
I left this in to remind myself to ask about `currentName` on the tokens to close objects. Have a look at what this prints in some of the tests. I'm not sure that `currentName` has the value you'd expect there. I don't *think* I use it in that context, but I wanted to bring it up.